src/main/java/fact/utils/CastDoubleArrayToIntArray.java

```diff
@@ -0,0 +1,40 @@
+package fact.utils;
+
+import stream.Data;
+import stream.Processor;
+import stream.annotations.Parameter;
+
+public class CastDoubleArrayToIntArray implements Processor {
+
+    @Parameter(
+        required = true,
+        description = "Key to your double array."
+    )
+    private String inputKey;
+
+    @Parameter(
+        required = true,
+        description = "Key to the output integer array."
+    )
+    protected String outputKey;
+
+    @Override
+    public Data process(Data input) {
+        double[] doubleArray = (double[]) input.get(inputKey);
+        int[] intArray = new int[doubleArray.length];
+
+        for (int i=0; i<intArray.length; ++i)
```

Comment:
Ah, sorry, wasn't heartbleed. It was `goto fail` https://blog.codecentric.de/en/2014/02/curly-braces/.

The missing braces thingy :) I just don't like `if`s and `for`s without braces